src/interfaces/pal.hpp

```diff
@@ -113,6 +113,48 @@ namespace pal
     constexpr size_t SHA1_HASH_SIZE = 20;
 
     bool ComputeSha1Hash(span<uint8_t const> data, std::array<uint8_t, SHA1_HASH_SIZE>& hashDestination);
+    
+    // A simple read-write lock that provides accessors to meet the C++11 BasicLockable requirements.
+    class ReadWriteLock;
+
+    class ReadLock final
```

Comment:
Should we explicitly delete the copy/move constructors for `ReadLock` and `WriteLock`?